src/platform/shared.c

```diff
@@ -1114,3 +1115,51 @@ unsigned int platform_get_ticks()
 {
 	return SDL_GetTicks();
 }
+
+uint8 platform_get_currency_value(char *currCode) {
+	if (currCode == NULL || strlen(currCode) < 3) {
+			return CURRENCY_POUNDS;
+	}
+	
+	if (strncmp(currCode, "GBP", 3) == 0){
+		return CURRENCY_POUNDS;
+	}
+	else if (strncmp(currCode, "USD", 3) == 0) {
+		return CURRENCY_DOLLARS;
+	}
+	else if (strncmp(currCode, "EUR", 3) == 0) {
+		return CURRENCY_EUROS;
+	}
+	else if (strncmp(currCode, "SEK", 3) == 0) {
+		return CURRENCY_KRONA;
+	}
+	else if (strncmp(currCode, "DEM", 3) == 0) {
+		return CURRENCY_DEUTSCHMARK;
+	}
+	else if (strncmp(currCode, "ITL", 3) == 0) {
+		return CURRENCY_LIRA;
+	}
+	else if (strncmp(currCode, "JPY", 3) == 0) {
+		return CURRENCY_YEN;
+	}
+	else if (strncmp(currCode, "ESP", 3) == 0) {
+		return CURRENCY_PESETA;
+	}
+	else if (strncmp(currCode, "FRF", 3) == 0) {
+		return CURRENCY_FRANC;
+	}
+	else if (strncmp(currCode, "NLG", 3) == 0) {
+		return CURRENCY_GUILDERS;
+	}
+	else if (strncmp(currCode, "KRW", 3) == 0) {
+		return CURRENCY_WON;
+	}
+	else if (strncmp(currCode, "RUB", 3) == 0) {
+		return CURRENCY_ROUBLE;
+	}
+	else if (strncmp(currCode, "CZK", 3) == 0) {
+		return CURRENCY_CZECH_KORUNA;
+	}
```

Comment:
Can we change all of this to a table instead?

C lacks a switch but if,else,if,else shouldn't be the solution. I have used tables in many other places, like command line, language, config etc. which you can refer to.